lib/components/searchbox.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import Mousetrap from 'mousetrap';
+
+import Component from '../component';
+
+class Searchform extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchTerm: '',
+      searchError: false
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleSearchTermChange = this.handleSearchTermChange.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    // If a submit button was selected, return focus back to the input box
+    this.searchBox.focus();
+
+    const term = this.props.terms && this.props.terms.getActiveTerm();
+    if (term) {
+      const searchTerm = this.state.searchTerm;
+      const options = {
+        dir: this.state.dir
+      };
+      const searchResult = term.find(searchTerm, options);
+      this.setState({searchError: !searchResult});
```

Comment:
Do you mean terminal wraparound (not line wrap-around?)

Currently the `find` method will always find the next item, and wraparound implicitly with no feedback to the user.

Ideally there should be a notification of this wraparound having happened to the user, but I want to address this in a separate PR